packages/native/src/useLinking.native.tsx

```diff
@@ -58,8 +58,9 @@ export default function useLinking(
 
   const extractPathFromURL = React.useCallback((url: string) => {
     for (const prefix of prefixesRef.current) {
-      if (url.startsWith(prefix)) {
-        return url.replace(prefix, '');
+      const prefixRegex = new RegExp(`^${prefix.replace(/\*/g, '[^/]+')}`);
+      if (url.match(prefixRegex)) {
```

Comment:
Since we're constructing the regex ourselves, `.test` is fine:
```suggestion
      if (prefixRegex.test(url)) {
```